File: solutions/Python/burst_balloons_greedy.py
from typing import List

def findMinArrowShots(points: List[List[int]]) -> int:
    """Greedy approach: O(n log n) time, O(n) space"""
    if not points:
        return 0
    points.sort(key=lambda x: x[1])
    arrows = 1
    last_end = points[0][1]
    for start, end in points[1:]:
        if start > last_end:
            arrows += 1
            last_end = end
    return arrows

File: solutions/C++/burst_balloons_greedy.cpp
#include <vector>
#include <algorithm>
using namespace std;

int findMinArrowShots(vector<vector<int>>& points) {
    if (points.empty()) return 0;
    sort(points.begin(), points.end(), [](const vector<int>& a, const vector<int>& b) {
        return a[1] < b[1];
    });
    int arrows = 1;
    long long last_end = points[0][1];
    for (int i = 1; i < points.size(); i++) {
        if (points[i][0] > last_end) {
            arrows++;
            last_end = points[i][1];
        }
    }
    return arrows;
}
